#14 implement full archive for syllabus

amiedavis opened the pull request:
Delivers #14

amiedavis committed:
#14 Implemented archive for syllabus using new pattern

amiedavis committed:
Merge branch 'master' into #14-implement-full-archive-for-syllabus

# Conflicts:
#	pom.xml

steveswinsburg commented on provider-syllabus/pom.xml:
```diff
@@ -0,0 +1,52 @@
+		<dependency>
+			<groupId>org.sakaiproject.archiver</groupId>
+			<artifactId>archiver-api</artifactId>
+			<version>0.2-SNAPSHOT</version>
```
FYI this version can now be removed as it is in the parent pom and will be inherited.

steveswinsburg commented on provider-syllabus/src/java/org/sakaiproject/archiver/provider/SyllabusArchiver.java:
```diff
@@ -0,0 +1,107 @@
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.ServerOverloadException;
```
is this actually a thing that this can throw?

amiedavis replied:
Yes

steveswinsburg replied:
Cool name.

steveswinsburg commented on provider-syllabus/src/java/org/sakaiproject/archiver/provider/SyllabusArchiver.java:
```diff
@@ -0,0 +1,107 @@
+					this.archiverService.archiveContent(archiveId, siteId, "sakai.syllabus", syllabusAttachmentBytes, syllabusAttachment.getName(), syllabus.getTitle());
+					log.debug("Attachment:  " +syllabusAttachment.getName() + " Sub-directory: " + syllabus.getTitle());
+				} catch (ServerOverloadException | PermissionException | IdUnusedException | TypeException e) {
+					e.printStackTrace();
```
Need to log this and continue, not chuck a stacktrace.

steveswinsburg commented on provider-syllabus/src/java/org/sakaiproject/archiver/provider/SyllabusArchiver.java:
```diff
@@ -0,0 +1,107 @@
+				}
+			}
+			log.debug("JSON: " +Jsonifier.toJson(archiveItem));
+			this.archiverService.archiveContent(archiveId, siteId, "sakai.syllabus", Jsonifier.toJson(archiveItem).getBytes(), archiveItem.getTitle() + ".json");
```
Jsonifier called twice, 83 and 84. Make a string var and then just use that.

amiedavis committed:
#14 Fixes as per PR comments

Remove archiver-api version from pom
Log error when getting attachments, and continue
Save content separately as html rather than json

amiedavis committed:
Merge branch 'master' into #14-implement-full-archive-for-syllabus

# Conflicts:
#	pom.xml

steveswinsburg commented:
Done in m1.
